### PR TITLE
SPLICE-1119 : Add support for Optional Compression 

### PR DESCRIPTION
**Pull request by jfilali**
SPLICE-1119 : Add support for Optional Compression for writing Parquet, ORC

If Compression is use with TextFILE, throw a exception

currently supporting zlib and snappy. lzo is not well supported by hadoop...

### CODE REVIEW

File: hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java [hunk 1]
```diff
@@ -673,7 +676,7 @@ private String planIfLast(AbstractSpliceFunction f, boolean isLast) {
             for (int i = 0; i < partitionBy.length; i++) {
                 partitionByCols.add(new Column(ValueRow.getNamedColumn(partitionBy[i])));
             }
-            insertDF.write().option("compression","none")
+            insertDF.write().option("compression",compression)
                     .partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
```
Comment (jyuanca) [inline hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java]: Define constant for "compression"

**Comment by jleach4** [comment]
+1

**Comment by jfilali** [comment]
@jyuanca : i fixed you comment

**Comment by jfilali** [comment]
do not merge code at this point , master is still broken, we need to fix
first master

2016-11-13 16:00 GMT-08:00 jyuanca <notifications@github.com>:

**Pull request merged**